### PR TITLE
Implement with proxy object

### PR DESCRIPTION
**Pull request by tkh44**
https://github.com/tkh44/retarget/issues/1

cc @FWeinb

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,97 +1,62 @@
...
-
-    if (obj == null) {
-      return obj
+import 'setimmediate';
+
+// Used to transfer object from a `toPrimitive` call
+// to the recieving retarget
+let transfer;
+
+const removeTransferOnNextTick = () => {
```
Comment (tkh44) [inline src/index.js]: Why is this needed?

**Comment by tkh44** [comment]
Is it possible to support the template literal syntax with the Proxy? 

I tried here and ran out of time. https://github.com/tkh44/retarget/tree/mixing-proxy-and-template-literals

File: src/index.js (hunk 1 above)
Comment (FWeinb) [reply src/index.js]: If for some reason someone would concat a selector function with a string or use any operator on it it add itself as a transfer. Because I am abusing `String.toPrimitve` to have this `retarget.a[retarget.b]` equal `retarget.a.b`. This trick would remove the transfer after the next tick. 

~~To change this I have an idea that would slightly alter the syntax for retarget composition. It looks like this:~~

~~`retarget.a(retarget.b) => retarget.a.b`~~

~~This solution would enable us to write a babel plugin that would transpile retarget proxies and make it work in any browser.~~

**Comment by FWeinb** [comment]
As I said above I found a better implementation for retarget with proxies ~~that would alter the compose syntax a bit~~ but would allow for a babel plugin. I don't know how feasible it would be to mix both syntaxes but It is possible.

**Comment by FWeinb** [comment]
I just optimised my initial implementation and removed the need for `setImmediate` and the whole `transfer` object. I will use this as a base to implement the tagged template syntax.

**Comment by FWeinb** [comment]
@tkh44  
I just finished adding the tagged template literal syntax to my proxy implementation, you can find it here: https://github.com/FWeinb/retarget/tree/proxy-and-template-literals
Reading the [diff](https://github.com/FWeinb/retarget/commit/f9d0d11f7f4aca0a3d26aa60d2e0d94a498fb07f#diff-1fdf421c05c1140f6d71444ea2b27638) is probably easer to understand. You can test it live in [this codesandbox](https://codesandbox.io/s/k1o112kz9o).

If these two ways of using retarget should be mixed is a whole other discussion.

**Comment by tkh44** [comment]
This is amazing! 😍
